aerogear/mobile-security-service-operator
# Refectory controller/reconcile code and fetch instances before updates in order to avoid issues regards the CR be already changed

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9169

## What
- Add export app namespace and remove unnecessary commands
- Refectory controller/reconcile code and fetch instances before updates in order to avoid issues regards the CR be already changed

## Why

**Explanation**
Logs as `Failed to update Status for the Component .... the object has been modified; please apply your changes to the latest version and try again"}` are faced because when the client will update the CR it was already changed. It is like when we go by the OCP UI and try to edit the YAML file of one CR which was changed between the time that the page was rendered and our click on the save button. 
- Try to avoid the above log issues
- Make easier troubleshooting and follow the logs.
- Make easier to keep the project maintained and allow it to be improved. 

## How
- Fetch the latest version of the CR before the updates
- Add comments to explain what is doing in each part
- Improve/Refactory the code 

## Verification Steps
- Install the operator with this PR and image `docker.io/cmacedo/mobile-security-service-operator:AEROGEAR-9169`
- Run `make create-all`
- Check that all was created with success
- In Resources >> other resources: Check that the status was update

<img width="1352" alt="Screenshot 2019-05-12 at 20 27 54" src="https://user-images.githubusercontent.com/7708031/57586903-83960500-74f4-11e9-9674-b6d7cf662159.png">

## Checklist:


## Progress


## Additional Notes
